consensus/hotstuff/forks/forks2.go

```diff
@@ -0,0 +1,557 @@
+package forks
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/forest"
+)
+
+// FinalityProof represents a finality proof for a Block. By convention, a FinalityProof
+// is immutable. Finality in Jolteon/HotStuff is determined by the 2-chain rule:
+//
+//	There exists a _certified_ block C, such that Block.View + 1 = C.View
+type FinalityProof struct {
+	Block          *model.Block
+	CertifiedChild model.CertifiedBlock
+}
+
+// Forks enforces structural validity of the consensus state and implements
+// finalization rules as defined in Jolteon consensus https://arxiv.org/abs/2106.10362
+// The same approach has later been adopted by the Diem team resulting in DiemBFT v4:
+// https://developers.diem.com/papers/diem-consensus-state-machine-replication-in-the-diem-blockchain/2021-08-17.pdf
+// Forks is NOT safe for concurrent use by multiple goroutines.
+type Forks2 struct {
+	notifier hotstuff.FinalizationConsumer
+	forest   forest.LevelledForest
+
+	trustedRoot          *model.CertifiedBlock
+	newestView           uint64 // newestView is the highest view of block proposal stored in Forks
```

Comment:
do we still need `newestView`?

Reply:
💡 that is a good point:
* We currently call the `NewestView()` in the EventHandler [here](https://github.com/onflow/flow-go/blob/0b531f40e749c12d206236dc196b2e6183f482b2/consensus/hotstuff/eventhandler/event_handler.go#L322)
* Though, you are right that this logic will be removed, since we don't depend on iterating over forks to re-populate the PaceMaker. 

removed in https://github.com/onflow/flow-go/pull/4190/commits/4c70d562ea01013fbbf1cbf079e6a38ed6fc04f7